llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -193,6 +194,7 @@ def build_inner_model(
         config_overrides: Dict[str, Any],
         load_in_8bit: bool,
         pretrained: bool,
+        model_cls: Union[AutoModel, PreTrainedModel] = AutoModelForCausalLM,
```

Comment:
technically `issubclass(AutoModelForCausalLM, AutoModel) == False`. I think this line should be
```suggestion
        model_cls: Union[Type, Type[PreTrainedModel]] = AutoModelForCausalLM,
```